### PR TITLE
`[Move]` Part-5 Classes into Different Library - `Neo.Extensions`

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Description
Reorganized and move classes from `neo` library to `neo.extensions` library. 

### Fixed problems and warnings that occurred. Along with naming violations.

# Change Log
 - Moved methods from `Neo.IO.Helper` class
 - Added `ICollectionExtensions`, `MemoryExtensions` classes
 - Added more methods to `StringExtensions`
 - Added `UnsafeData` class for some methods _(non-extensions)_

## Type of change

- [x] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- Unit Testing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo.Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -32,5 +32,17 @@ public static byte[] HexToBytes(this string value)
...
+        /// Gets the size of the specified <see cref="string"/> encoded in variable-length encoding.
+        /// </summary>
+        /// <param name="value">The specified <see cref="string"/>.</param>
+        /// <returns>The size of the <see cref="string"/>.</returns>
+        public static int GetVarSize(this string value)
+        {
+            var size = Utility.StrictUTF8.GetByteCount(value);
+            return UnsafeData.GetVarSize(size) + size;
+        }
+
```
Comment (shargon) [inline src/Neo.Extensions/StringExtensions.cs:46]:
```suggestion
```

File: src/Neo/Network/P2P/Payloads/Conditions/OrCondition.cs [hunk 2]
```diff
@@ -30,7 +31,14 @@ public class OrCondition : WitnessCondition
         /// </summary>
         public WitnessCondition[] Expressions;
 
-        public override int Size => base.Size + Expressions.GetVarSize();
+        public override int Size
```
Comment (shargon) [inline src/Neo/Network/P2P/Payloads/Conditions/OrCondition.cs:34]: This change seems unnecessary

File: tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ namespace Neo.Extensions.Tests
     [TestClass]
     public class UT_BigIntegerExtensions
     {
+
```
Comment (shargon) [inline tests/Neo.Extensions.Tests/UT_BigIntegerExtensions.cs:21]:
```suggestion
```

File: tests/Neo.UnitTests/IO/UT_IOHelper.cs [hunk 4]
```diff
@@ -110,23 +110,28 @@ public void TestNullableArray()
...
-                    ISerializable result = Neo.IO.Helper.AsSerializable(caseArray, typeof(UInt160));
-                    Assert.AreEqual(UInt160.Zero, result);
-                }
-                else
-                {
-                    Action action = () => Neo.IO.Helper.AsSerializable(Array.Empty<byte>(), typeof(double));
-                    action.Should().Throw<InvalidCastException>();
-                }
-            }
+            byte[] caseArray = [0x00,
```
Comment (shargon) [inline tests/Neo.UnitTests/IO/UT_IOHelper.cs:113]: In less lines is better

File: tests/Neo.Extensions.Tests/UT_StringExtensions.cs [hunk 5]
```diff
@@ -35,5 +35,164 @@ public void TestHexToBytes()
...
+                    Assert.AreEqual(3, result);
+                }
+                else
+                {
+                    int result = UnsafeData.GetVarSize(0xFFFFFF);
+                    Assert.AreEqual(5, result);
+                }
+            }
+        }
+
```
Comment (shargon) [inline tests/Neo.Extensions.Tests/UT_StringExtensions.cs:68]:
```suggestion
```

**Review by shargon — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**